app/scripts/lib/mmi-hooks.ts

```diff
@@ -0,0 +1,66 @@
+import { TransactionMeta } from '../controllers/transactions/tx-state-manager';
+
+/**
+ * Whether or not to skip publishing the transaction.
+ *
+ * @param txMeta - The transaction meta.
+ * @param signedEthTx - Signed Ethereum transaction.
+ * @param addTransactionToWatchList
+ */
+export function afterTransactionSign(
+  txMeta: TransactionMeta,
+  signedEthTx: any,
+  addTransactionToWatchList: (
+    custodianTransactionId: string | undefined,
+    from?: string,
+    bufferType?: string,
+    isSignedMessage?: boolean,
+  ) => Promise<void>,
+): boolean {
+  // MMI does not broadcast transactions, as that is the responsibility of the custodian
+  if (txMeta?.custodyStatus) {
+    txMeta.custodyId = signedEthTx.custodian_transactionId;
+    txMeta.custodyStatus = signedEthTx.transactionStatus;
+    addTransactionToWatchList(txMeta.custodyId, txMeta.txParams.from);
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Whether or not should run logic before publishing the transaction.
+ *
+ * @param txMeta - The transaction meta.
+ */
+export function beforeTransactionPublish(txMeta: TransactionMeta): boolean {
+  // MMI does not broadcast transactions, as that is the responsibility of the custodian
+  if (txMeta?.custodyStatus) {
```

Comment:
Very minor, but should we simplify to the following?

```
return !txMeta?.custodyStatus;
```

Same for `beforeTransactionApproveOnInit`?

Reply:
Perfect, done on both functions.